pkg/config/database.go

```diff
@@ -35,30 +41,73 @@ func (d *Database) Open(logger *logging.Logger) (*icingadb.DB, error) {
 		driver.Register(logger)
 	})
 
-	config := mysql.NewConfig()
+	var dsn string
+	switch d.Type {
+	case "mysql":
+		config := mysql.NewConfig()
 
-	config.User = d.User
-	config.Passwd = d.Password
-	config.Net = "tcp"
-	config.Addr = net.JoinHostPort(d.Host, fmt.Sprint(d.Port))
-	config.DBName = d.Database
-	config.Timeout = time.Minute
+		config.User = d.User
+		config.Passwd = d.Password
+		config.Net = "tcp"
+		config.Addr = net.JoinHostPort(d.Host, fmt.Sprint(d.Port))
+		config.DBName = d.Database
+		config.Timeout = time.Minute
+		config.Params = map[string]string{"sql_mode": "ANSI_QUOTES"}
 
-	tlsConfig, err := d.TLS.MakeConfig(config.Addr)
-	if err != nil {
-		return nil, err
-	}
+		tlsConfig, err := d.TLS.MakeConfig(config.Addr)
+		if err != nil {
+			return nil, err
+		}
 
-	if tlsConfig != nil {
-		config.TLSConfig = "icingadb"
-		if err := mysql.RegisterTLSConfig(config.TLSConfig, tlsConfig); err != nil {
-			return nil, errors.Wrap(err, "can't register TLS config")
+		if tlsConfig != nil {
+			config.TLSConfig = "icingadb"
+			if err := mysql.RegisterTLSConfig(config.TLSConfig, tlsConfig); err != nil {
+				return nil, errors.Wrap(err, "can't register TLS config")
+			}
+		}
+
+		dsn = config.FormatDSN()
+	case "pgsql":
+		uri := &url.URL{
+			Scheme:   "postgres",
+			User:     url.UserPassword(d.User, d.Password),
+			Host:     net.JoinHostPort(d.Host, strconv.FormatInt(int64(d.Port), 10)),
+			Path:     "/" + url.PathEscape(d.Database),
+			RawQuery: "connect_timeout=60&binary_parameters=yes",
```

Comment:
I think using `url.Values` makes query string manipulation much cleaner to read.